hail/src/main/scala/is/hail/expr/types/physical/PCanonicalStruct.scala

```diff
@@ -50,10 +52,11 @@ final case class PCanonicalStruct(fields: IndexedSeq[PField], required: Boolean
   }
 
   val missingIdx = new Array[Int](size)
+  // sets missingIdx indices
```

Comment:
I found this confusing. In a later PR I would like to refactor this. This pattern of mutating a passed array is unnecessary indirection, and a bit unusual for our codebase (in which functions are idempotent).

Reply:
fixed in the ptypes parser PR, as you know.